examples/nlp/gpt/conf/training_rm.yaml

```diff
@@ -63,10 +63,12 @@ model:
     merge_attributes: False # whether to merge multiple attributes into a scalar
     attribute_weights: null # apply these weights to each attributes when merging them into a scalar
     loss_mask_val: -100 #  mask dimensions with this value when calculating MSE loss
+    loss_func: regression # ["regression", "regular_bt", "margin_bt", "scaled_bt"]
+    load_rm_head_weights: False # [False, True] False only loads base model while True loads rm_head weights as well (useful for intializing rm_head with model containing existing rm_head)
   output_sequence: False  # Whether to output a single scalar or a sequence of scalars.
   use_avg_pool: False  # Whether to use avg pool to sum across the sequence dim in reward model
-  force_head_dtype: float32  # enforce specific dtype for the final projection in the model head
-  micro_batch_size: 1
+  force_head_dtype: bfloat16 #float32  # enforce specific dtype for the final projection in the model head
```

Comment:
@HeyyyyyyG I want to give you a heads up that this has been changed to bfloat16

The main reason is the CI doesn't pass with float32 (likely due to newer mcore version). Also float32 for just the reward head (everywhere else is bfloat16) isn't supported by TRTLLM or HF compatible checkpoint. My test of post-training-converted bfloat16 weights shows little degradation so I think it's ok to keep it as bfloat16.

```
[rank1]:   File "/opt/NeMo-Aligner/nemo_aligner/algorithms/supervised.py", line 210, in fit
[rank1]:     loss, metrics = self.train_single_step(batch)
[rank1]:   File "/opt/NeMo-Aligner/nemo_aligner/algorithms/supervised.py", line 145, in train_single_step
[rank1]:     loss_mean, metrics = self.model.get_loss_and_metrics(batch=batch, forward_only=False)
[rank1]:   File "/opt/NeMo-Aligner/nemo_aligner/models/nlp/gpt/megatron_gpt_regression_reward_model.py", line 182, in get_loss_and_metrics
[rank1]:     losses_reduced_per_micro_batch = fwd_bwd_function(
[rank1]:   File "/opt/Megatron-LM/megatron/core/pipeline_parallel/schedules.py", line 474, in forward_backward_no_pipelining
[rank1]:     backward_step(input_tensor, output_tensor, output_tensor_grad, model_type, config)
[rank1]:   File "/opt/Megatron-LM/megatron/core/pipeline_parallel/schedules.py", line 354, in backward_step
[rank1]:     custom_backward(output_tensor[0], output_tensor_grad[0])
[rank1]:   File "/opt/Megatron-LM/megatron/core/pipeline_parallel/schedules.py", line 154, in custom_backward
[rank1]:     Variable._execution_engine.run_backward(
[rank1]:   File "/usr/local/lib/python3.10/dist-packages/torch/autograd/function.py", line 301, in apply
[rank1]:     return user_fn(self, *args)
[rank1]:   File "/usr/local/lib/python3.10/dist-packages/torch/cuda/amp/autocast_mode.py", line 142, in decorate_bwd
[rank1]:     return bwd(*args, **kwargs)
[rank1]:   File "/opt/Megatron-LM/megatron/core/tensor_parallel/layers.py", line 497, in backward
[rank1]:     fused_weight_gradient_mlp_cuda.wgrad_gemm_accum_fp16(
[rank1]: RuntimeError: expected scalar type BFloat16 but found Float
```

Reply:
see this CI https://github.com/NVIDIA/NeMo-Aligner/actions/runs/11565227336/job/32191971677#step:5:716

Reply:
cc @odelalleau

Reply:
but we'd be training with BF16 head and I believe there was an issue with this before

Reply:
@gshennvm What was the issue?

Reply:
I believe it was convergence but @odelalleau ran those experiments so best for him to give details there

Reply:
After discussion with @gshennvm and @odelalleau , we decided to keep it as BF16 by default.